Building errors with nextcloud/libcloudproviders-integration

Pulled the master up to [nextcloud/libcloudproviders-integration](https://github.com/nextcloud/client/commit/de4492a2fd7cb9134392b89453b02c8719a78755) and got building errors. Hope this helps.

I'm building a [Windows Installer Build](https://doc.owncloud.org/desktop/2.4/building.html#windows-installer-build-cross-compile) within a fresh docker environment. Just following the document.
```
git clone https://github.com/frankosterfeld/qtkeychain.git
git clone git://github.com/owncloud/client.git
cd client
git submodule init
git submodule update

cd admin/win/docker/
docker build . -t nextcloud-client-win32:latest
docker create -ti --name nc nextcloud-client-win32:latest /bin/bash
docker start nc
```

Applied a patch to avoid the header not found error.
```
diff --git a/src/gui/owncloudgui.h b/src/gui/owncloudgui.h
index f1341545f..4c5eb1f52 100644
--- a/src/gui/owncloudgui.h
+++ b/src/gui/owncloudgui.h
@@ -25,7 +25,7 @@
 #include <QMenu>
 #include <QSize>
 #include <QTimer>
-#include <QDBusConnection>
+#include <QtDBus/QDBusConnection>

 namespace OCC {
```

Tried to build.
```
cd ..
docker cp client nc:/home/user/
docker cp qtkeychain nc:/home/user/
docker exec nc chmod 777 /home/user/client
docker exec nc chmod 777 /home/user/qtkeychain

docker exec nc /home/user/client/admin/win/docker/build.sh client
```

And got errors.
```
[ 33%] Building CXX object src/gui/CMakeFiles/nextcloud.dir/main.cpp.obj
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/oleidl.h:7:0,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/ole2.h:38,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/wtypes.h:12,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/winscard.h:10,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/windows.h:97,
                 from /home/user/client/src/csync/std/c_private.h:34,
                 from /home/user/client/src/csync/csync.h:35,
                 from /home/user/client/src/gui/../libsync/syncfileitem.h:24,
                 from /home/user/client/src/gui/../libsync/syncresult.h:23,
                 from /home/user/client/src/gui/application.h:27,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:150:78: error: expected ',' or '...' before 'struct'
     bool connect(const QString &service, const QString &path, const QString &interface,
                                                                              ^
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:152:78: error: expected ',' or '...' before 'struct'
     bool connect(const QString &service, const QString &path, const QString &interface,
                                                                              ^
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/QDBusConnection:1:0,
                 from /home/user/client/src/gui/owncloudgui.h:28,
                 from /home/user/client/src/gui/application.h:29,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:152:10: error: 'bool QDBusConnection::connect(const QString&, const QString&, const QString&)' cannot be overloaded
     bool connect(const QString &service, const QString &path, const QString &interface,
          ^~~~~~~
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:150:10: error: with 'bool QDBusConnection::connect(const QString&, const QString&, const QString&)'
     bool connect(const QString &service, const QString &path, const QString &interface,
          ^~~~~~~
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/oleidl.h:7:0,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/ole2.h:38,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/wtypes.h:12,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/winscard.h:10,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/windows.h:97,
                 from /home/user/client/src/csync/std/c_private.h:34,
                 from /home/user/client/src/csync/csync.h:35,
                 from /home/user/client/src/gui/../libsync/syncfileitem.h:24,
                 from /home/user/client/src/gui/../libsync/syncresult.h:23,
                 from /home/user/client/src/gui/application.h:27,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:155:78: error: expected ',' or '...' before 'struct'
     bool connect(const QString &service, const QString &path, const QString &interface,
                                                                              ^
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/QDBusConnection:1:0,
                 from /home/user/client/src/gui/owncloudgui.h:28,
                 from /home/user/client/src/gui/application.h:29,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:155:10: error: 'bool QDBusConnection::connect(const QString&, const QString&, const QString&)' cannot be overloaded
     bool connect(const QString &service, const QString &path, const QString &interface,
          ^~~~~~~
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:150:10: error: with 'bool QDBusConnection::connect(const QString&, const QString&, const QString&)'
     bool connect(const QString &service, const QString &path, const QString &interface,
          ^~~~~~~
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/oleidl.h:7:0,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/ole2.h:38,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/wtypes.h:12,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/winscard.h:10,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/windows.h:97,
                 from /home/user/client/src/csync/std/c_private.h:34,
                 from /home/user/client/src/csync/csync.h:35,
                 from /home/user/client/src/gui/../libsync/syncfileitem.h:24,
                 from /home/user/client/src/gui/../libsync/syncresult.h:23,
                 from /home/user/client/src/gui/application.h:27,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:159:81: error: expected ',' or '...' before 'struct'
     bool disconnect(const QString &service, const QString &path, const QString &interface,
                                                                                 ^
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:161:81: error: expected ',' or '...' before 'struct'
     bool disconnect(const QString &service, const QString &path, const QString &interface,
                                                                                 ^
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/QDBusConnection:1:0,
                 from /home/user/client/src/gui/owncloudgui.h:28,
                 from /home/user/client/src/gui/application.h:29,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:161:10: error: 'bool QDBusConnection::disconnect(const QString&, const QString&, const QString&)' cannot be overloaded
     bool disconnect(const QString &service, const QString &path, const QString &interface,
          ^~~~~~~~~~
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:159:10: error: with 'bool QDBusConnection::disconnect(const QString&, const QString&, const QString&)'
     bool disconnect(const QString &service, const QString &path, const QString &interface,
          ^~~~~~~~~~
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/oleidl.h:7:0,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/ole2.h:38,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/wtypes.h:12,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/winscard.h:10,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/windows.h:97,
                 from /home/user/client/src/csync/std/c_private.h:34,
                 from /home/user/client/src/csync/csync.h:35,
                 from /home/user/client/src/gui/../libsync/syncfileitem.h:24,
                 from /home/user/client/src/gui/../libsync/syncresult.h:23,
                 from /home/user/client/src/gui/application.h:27,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:164:81: error: expected ',' or '...' before 'struct'
     bool disconnect(const QString &service, const QString &path, const QString &interface,
                                                                                 ^
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/QDBusConnection:1:0,
                 from /home/user/client/src/gui/owncloudgui.h:28,
                 from /home/user/client/src/gui/application.h:29,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:164:10: error: 'bool QDBusConnection::disconnect(const QString&, const QString&, const QString&)' cannot be overloaded
     bool disconnect(const QString &service, const QString &path, const QString &interface,
          ^~~~~~~~~~
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:159:10: error: with 'bool QDBusConnection::disconnect(const QString&, const QString&, const QString&)'
     bool disconnect(const QString &service, const QString &path, const QString &interface,
          ^~~~~~~~~~
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/oleidl.h:7:0,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/ole2.h:38,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/wtypes.h:12,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/winscard.h:10,
                 from /usr/i686-w64-mingw32/sys-root/mingw/include/windows.h:97,
                 from /home/user/client/src/csync/std/c_private.h:34,
                 from /home/user/client/src/csync/csync.h:35,
                 from /home/user/client/src/gui/../libsync/syncfileitem.h:24,
                 from /home/user/client/src/gui/../libsync/syncresult.h:23,
                 from /home/user/client/src/gui/application.h:27,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:170:61: error: expected ',' or '...' before 'struct'
     bool registerObject(const QString &path, const QString &interface, QObject *object,
                                                             ^
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:181:31: error: expected unqualified-id before 'struct'
     QDBusConnectionInterface *interface() const;
                               ^
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/QDBusConnection:1:0,
                 from /home/user/client/src/gui/owncloudgui.h:28,
                 from /home/user/client/src/gui/application.h:29,
                 from /home/user/client/src/gui/main.cpp:24:
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:181:30: error: expected ';' at end of member declaration
     QDBusConnectionInterface *interface() const;
                              ^
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:181:40: error: expected identifier before '(' token
     QDBusConnectionInterface *interface() const;
                                        ^
/usr/i686-w64-mingw32/sys-root/mingw/include/qt5/QtDBus/qdbusconnection.h:181:41: error: expected unqualified-id before ')' token
     QDBusConnectionInterface *interface() const;
                                         ^
cc1plus: warning: unrecognized command line option '-Wno-gnu-zero-variadic-macro-arguments'
make[2]: *** [src/gui/CMakeFiles/nextcloud.dir/main.cpp.obj] Error 1
make[1]: *** [src/gui/CMakeFiles/nextcloud.dir/all] Error 2
make: *** [all] Error 2
src/gui/CMakeFiles/nextcloud.dir/build.make:403: recipe for target 'src/gui/CMakeFiles/nextcloud.dir/main.cpp.obj' failed
CMakeFiles/Makefile2:439: recipe for target 'src/gui/CMakeFiles/nextcloud.dir/all' failed
Makefile:151: recipe for target 'all' failed
```

--- a/src/gui/owncloudgui.cpp
+++ b/src/gui/owncloudgui.cpp
@@ -42,8 +42,10 @@
 #include <QDir>
 #include <QMessageBox>
 #include <QSignalMapper>
+#ifdef WITH_LIBCLOUDPROVIDERS
 #include <QtDBus/QDBusConnection>
 #include <QtDBus/QDBusInterface>
+#endif
 
 #if defined(Q_OS_X11)
 #include <QX11Info>
@@ -66,7 +68,9 @@ ownCloudGui::ownCloudGui(Application *parent)
 #endif
     _logBrowser(0)
     , _contextMenuVisibleOsx(false)
+#ifdef WITH_LIBCLOUDPROVIDERS
     , _bus(QDBusConnection::sessionBus())
+#endif
     , _recentActionsMenu(0)
     , _qdbusmenuWorkaround(false)
     , _app(parent)

--- a/src/gui/owncloudgui.h
+++ b/src/gui/owncloudgui.h
@@ -25,7 +25,9 @@
 #include <QMenu>
 #include <QSize>
 #include <QTimer>
+#ifdef WITH_LIBCLOUDPROVIDERS
 #include <QDBusConnection>
+#endif
 
 namespace OCC {
 
@@ -128,7 +130,9 @@ private:
     // on OSX because aboutToHide is not reliable everywhere.
     bool _contextMenuVisibleOsx;
 
+#ifdef WITH_LIBCLOUDPROVIDERS
     QDBusConnection _bus;
+#endif
 
     QMenu *_recentActionsMenu;
     QVector<QMenu *> _accountMenus;

